src/esp/gfx/Simulator.cpp

```diff
@@ -139,13 +139,23 @@ void Simulator::reconfigure(const SimulatorConfiguration& cfg) {
 
   semanticScene_ = nullptr;
   semanticScene_ = scene::SemanticScene::create();
-  if (io::exists(houseFilename)) {
-    scene::SemanticScene::loadMp3dHouse(houseFilename, *semanticScene_);
-  }
-
-  // also load SemanticScene for SUNCG house file
-  if (sceneInfo.type == assets::AssetType::SUNCG_SCENE) {
-    scene::SemanticScene::loadSuncgHouse(sceneFilename, *semanticScene_);
+  switch (sceneInfo.type) {
+    case assets::AssetType::INSTANCE_MESH:
+      houseFilename = io::parentPath(houseFilename) + "info_semantic.json";
```

Comment:
https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Utility_1_1Directory.html#a222baead7128ec0cadc433e3a8c0059f

Reply:
Then you can use `Cr::Directory::join` to combine the path with the `info_semantic.json` name.

Reply:
Done.